UN-OCHA/unocha-site
# UNO-601 E2e Tests

Recent base theme updates to e2e tests in the sub theme https://github.com/UN-OCHA/common_design/pull/431, adjusted for UNO UI components like Main menu, mega menu and search.

#To Test - Optional
1. In the `common_design_subtheme` run `npm i` to install node modules including Puppeteer, Jest and Chromium browser
2. Update `html/themes/contrib/common_design/common_design_subtheme/_tests/_env/index.js` if needed - Set 
` 'local': {  baseUrl: 'https://commondesign.test' `
3. Run `npm run e2e` to run the tests and/or `npm run e2e -- --debug` to see the steps in a Chromium instance.
4. Tests should all pass.
```
  console.log
    E2E baseUrl:  https://unocha-local.test

      at Object.log (_tests/_DebugEnv.test.js:5:13)

 PASS  _tests/_DebugEnv.test.js
  DebugEnv
    ✓ should output the environment that Jest is testing (21 ms)

 PASS  _tests/MainMenuDropdown.test.js (6.352 s)
  MainMenuDropdown
    ✓ should expand when clicked (414 ms)

 PASS  _tests/FooterMenu.test.js (6.36 s)
  FooterLinks
    ✓ should contain specific links in the Footer menu (459 ms)

 PASS  _tests/SearchDropdown.test.js (7.077 s)
  SearchDropdown
    ✓ should expand when clicked (392 ms)

 PASS  _tests/HeadingOne.test.js (7.832 s)
  HeadingOne
    ✓ should have a single heading one (235 ms)

 PASS  _tests/MobileMenu.test.js (9.673 s)
  MobileMenu
    ✓ should expand when clicked (368 ms)
    ✓ should expand when clicked (255 ms)
    ✓ should expand when clicked (263 ms)
    ✓ should expand when clicked (229 ms)

 PASS  _tests/MegaMenuDropdown.test.js (10.027 s)
  MegaMenuDropdown
    ✓ should expand when clicked (417 ms)

Test Suites: 7 passed, 7 total
Tests:       10 passed, 10 total
Snapshots:   0 total
Time:        10.476 s
Ran all test suites matching /_tests\/FooterMenu.test.js|_tests\/HeadingOne.test.js|_tests\/MainMenuDropdown.test.js|_tests\/MegaMenuDropdown.test.js|_tests\/MobileMenu.test.js|_tests\/SearchDropdown.test.js|_tests\/_DebugEnv.test.js/i.
```
